src/trix/models/composition.coffee

```diff
@@ -211,7 +219,13 @@ class Trix.Composition extends Trix.BasicObject
     @currentAttributes[attributeName]?
 
   toggleCurrentAttribute: (attributeName) ->
-    if value = not @currentAttributes[attributeName]
+    if ("header" in @getBlock()?.getAttributes() and 
+    attributeName in @getBlock()?.getAttributes() and
+    attributeName != "header")
```

Comment:
Yes, this is a bit confusing. The conditional is true when:

1.   The attribute associated with the button clicked is currently active. IE: Bullets is clicked twice. The conditional would be true on the second click.
2.   The current block also contains a header block.

If our the current block attributes are `Attributes: ["bulletList","bullet","header"]` and bullet is clicked,  `header` will be removed, `bullet` will be removed, and then `header` will be re-added. If this conditional is true, we can be sure that the last block attribute is header, since new block attributes can't be added after a header is added.